apps/webapp/eaas-client/src/ui/sarcophagi/ClientSarcophagi.tsx

```diff
@@ -27,18 +31,18 @@ export function ClientSarcophagi() {
           setIsLoadingSarcophagi(false);
         });
     }
-  }, [loadedSarcophagi, toast]);
+  }, [dispatch, loadedSarcophagi, toast]);
 
   function sarcophagiPanel() {
-    if (isLoadingSarcophagi) {
+    if (clientSarcophagi.length === 0 && isLoadingSarcophagi) {
```

Comment:
Why do you need the length check here? Seems like this could be removed

Reply:
This is because `isLoadingSarcophagi` does get set back to `true` when re-fetching, but since there's already sarcopahgi in the store, no need to show the loaded.